### ISSUE TITLE
False negative in LeftCurly for types with option=eol

### ISSUE THREAD

**Issue by Atharv3221** [checkstyle/checkstyle#20963]
I have read check documentation: https://checkstyle.sourceforge.io/checks/blocks/leftcurly.html#LeftCurly
I have read https://checkstyle.sourceforge.io/property_types.html#LeftCurlyOption
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Using confg:
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="LeftCurly">
      <property name="option" value="eol"/>
    </module>
  </module>
</module>
```

Types
```java
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat Test.java
public class Test { int a = 1;      // expected violation
}

interface Test1 { int b = 1;        // expected violation
}

@interface Test2 { int c = 1;       // expected violation
}

record Test3() { static int k = 0;  // expected violation
}
```
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-14.1.0-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```
Expected:
```
Starting audit...
[ERROR] /home/atharv/test/Test.java:1:19: '{' at column 19 should have line break after. [LeftCurly]
[ERROR] /home/atharv/test/Test.java:4:17: '{' at column 17 should have line break after. [LeftCurly]
[ERROR] /home/atharv/test/Test.java:7:18: '{' at column 18 should have line break after. [LeftCurly]
[ERROR] /home/atharv/test/Test.java:10:16: '{' at column 16 should have line break after. [LeftCurly]
Audit done.
Checkstyle ends with 4 errors.
```
---

Switch
```java
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat Test.java
public class Test {
    void method() {
        int a = 1;
        switch (a) {default: a+= 1;} // expected violation
    }
}
```
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-14.1.0-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```
Expected:
```
Starting audit...
[ERROR] /home/atharv/test/Test.java:4:20: '{' at column 20 should have line break after. [LeftCurly]
Audit done.
Checkstyle ends with 1 errors.
```
---

Anonymous Class
```java
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat Test.java
public class Test {

    void method() {
        Runnable runnable = new Runnable() { @Override
            public void run() {
            }
        };
    }
}
```
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-14.1.0-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```
Expected:
```
Starting audit...
[ERROR] /home/atharv/test/Test.java:4:45: '{' at column 45 should have line break after. [LeftCurly]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by Atharv3221** [comment]
I am working on this

**Comment by Atharv3221** [comment]
Anonymous Class violation is expected because, violation is reported for option [nl](https://checkstyle.sourceforge.io/property-types.html#LeftCurlyOption-nl)

With config;
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="LeftCurly">
      <property name="option" value="nl"/>
    </module>
  </module>
</module>
```
```java
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat Test.java
public class Test 
{

    void method() 
    {
        Runnable runnable = new Runnable() {
            @Override
            public void run() 
            {
            }
        };
    }
}
```
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-14.1.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/atharv/test/Test.java:6:44: '{' at column 44 should be on a new line. [LeftCurly]
Audit done.
Checkstyle ends with 1 errors
```

**Comment by romani** [comment]
@Atharv3221 , until end of summer do not work on non-google tickets.
lets finish Google style update as quick as possible.
